R/tm_a_pca.R

```diff
@@ -1,54 +1,61 @@
 #' Principal component analysis module
-#' @md
+#'
+#' Module conducts principal component analysis (PCA) on a given dataset and offers different
+#' ways of visualizing the outcomes, including elbow plot, circle plot, biplot, and eigenvector plot.
+#' Additionally, it enables dynamic customization of plot aesthetics, such as opacity, size, and
+#' font size, through UI inputs.
 #'
 #' @inheritParams teal::module
 #' @inheritParams shared_params
 #' @param dat (`data_extract_spec` or `list` of multiple `data_extract_spec`)
-#'   Columns used to compute PCA.
-#' @param alpha optional, (`numeric`) If scalar then the plot points will have a fixed opacity. If a
-#'   slider should be presented to adjust the plot point opacity dynamically then it can be a vector of
-#'   length three with `c(value, min, max)`.
-#' @param size optional, (`numeric`) If scalar then the plot point sizes will have a fixed size.
-#'   If a slider should be presented to adjust the plot point sizes dynamically then it can be a
-#'   vector of length three with `c(value, min, max)`.
-#' @param font_size optional, (`numeric`) font size control for title, x-axis label, y-axis label and legend.
-#'   If scalar then the font size will have a fixed size. If a slider should be presented to adjust the plot
-#'   point sizes dynamically then it can be a vector of length three with `c(value, min, max)`.
-#'
+#'   specifying columns used to compute PCA.
+#' @param alpha (`numeric`, optional) value or a vector of length three in the form `c(value, min, max)`.
+#' - If provided as a scalar, it sets a fixed opacity for plot points.
```

Comment:
@kartikeyakirar I don't think we have `scalars` in R. Maybe differentiate by `length`.

> If provided o `length==1`, it sets a fixed...
> If provided as a vector of `length>1`, it allows dynamic...

Reply:
Yeah, I know `checkmate::assert_scalar` exists, but I don't think we have scalars in R. Unsure if `scalar` is common for users who never seen `checkmate`. I would strive from using it in terms of R documenation if possible and would substitute with something else, just in case

Reply:
![Uploading image.png…]()

Reply:
<img width="948" alt="image" src="https://github.com/insightsengineering/teal.modules.general/assets/133694481/48db2d2d-c2b3-4f7c-8a1b-3328f1b28453">

Reply:
done! [057ae53](https://github.com/insightsengineering/teal.modules.general/pull/646/commits/057ae536febdf3ce45c4f7333e080fb3dc844c00)